src/gazebo_mavlink_interface.cpp

```diff
@@ -1508,7 +1508,7 @@ void GazeboMavlinkInterface::handle_control(double _dt)
 {
   // set joint positions
   for (int i = 0; i < input_reference_.size(); i++) {
-    if (joints_[i]) {
+    if (joints_[i] || joint_control_type_[i] == "position_gztopic") {
```

Comment:
I think a better approach is move the `ifelse` block at line 1533 to an `ifelse` at line 1558. Otherwise, the comment `        gzerr << "joint_control_type[" << joint_control_type_[i] << "] undefined.\n";` doesn't make sense

Reply:
I'm not sure... at line 1511 we are entering this block only if joint is set.
I added another case - when control type is 'position_gztopic' regardless of joint is set or not.

Reply:
Yes, my comment was why not just add it to the

Reply:
Ah, I missed that part.

Reply:
Right, I missed that part